src/internal/connector/graph/collections.go

```diff
@@ -51,28 +51,36 @@ func BaseCollections(
 	service path.ServiceType,
 	categories map[path.CategoryType]struct{},
 	su support.StatusUpdater,
-	errs *fault.Errors,
-) ([]data.BackupCollection, *fault.Errors) {
-	res := []data.BackupCollection{}
+) ([]data.BackupCollection, error) {
+	var (
+		errs = []error{}
+		res  = []data.BackupCollection{}
+	)
 
 	for cat := range categories {
 		p, err := path.Builder{}.Append("tmp").ToDataLayerPath(tenant, user, service, cat, false)
 		if err != nil {
 			// Shouldn't happen.
-			errs.Fail(clues.Wrap(err, "making path").WithAll("service", service, "category", cat))
+			errs = append(
+				errs,
+				clues.Wrap(err, "making path").WithAll("service", service, "category", cat))
+
 			continue
 		}
 
 		p, err = p.Dir()
 		if err != nil {
 			// Shouldn't happen.
-			errs.Fail(clues.Wrap(err, "getting base prefix").WithAll("serivce", service, "category", cat))
+			errs = append(
+				errs,
+				clues.Wrap(err, "getting base prefix").WithAll("serivce", service, "category", cat))
+
 			continue
 		}
 
 		// Pop off the last path element because we just want the prefix.
 		res = append(res, emptyCollection{p: p, su: su})
 	}
 
-	return res, errs
+	return res, clues.Stack(errs...)
```

Comment:
this will always return a non-nil value, unfortunately.  So you'll want to check for errors separately.
```suggestion
        if len(errs) > 0 {
            return clues.Stack(errs...)
        }
        
	return res, nil
```

We'll need an `ErrorOrNil()` func we want to collapse that in the future.  Not my favorite pattern, but it's doable.